tests/datasource/fluent/integration/test_sql_datasources.py

```diff
@@ -604,40 +603,180 @@ def test_checkpoint_run(
         "postgres",
         "snowflake",
         "sqlite",
     ],
+    'expect_column_values_to_not_be_null-str "unquoted_lower_col"': [
+        "postgres",
+        "snowflake",
+    ],
+    "expect_column_values_to_not_be_null-str UNQUOTED_LOWER_COL": [
+        "databricks_sql",
+        "postgres",
+        "sqlite",
+    ],
+    'expect_column_values_to_not_be_null-str "UNQUOTED_LOWER_COL"': [
+        "postgres",
+        "snowflake",
+    ],
+    "expect_column_values_to_not_be_null-str unquoted_upper_col": [
+        "databricks_sql",
+        "sqlite",
+    ],
+    'expect_column_values_to_not_be_null-str "unquoted_upper_col"': [
+        "postgres",
+        "snowflake",
+    ],
+    "expect_column_values_to_not_be_null-str UNQUOTED_UPPER_COL": [
+        "databricks_sql",
+        "postgres",
+    ],
+    'expect_column_values_to_not_be_null-str "UNQUOTED_UPPER_COL"': [
+        "postgres",
+        "snowflake",
+    ],
+    "expect_column_values_to_not_be_null-str quoted_lower_col": [
+        "postgres",
+        "snowflake",
+    ],
+    'expect_column_values_to_not_be_null-str "quoted_lower_col"': [
+        "postgres",
+        "snowflake",
+    ],
+    "expect_column_values_to_not_be_null-str QUOTED_LOWER_COL": [
+        "databricks_sql",
+        "postgres",
+        "snowflake",
+        "sqlite",
+    ],
+    'expect_column_values_to_not_be_null-str "QUOTED_LOWER_COL"': [
+        "postgres",
+        "snowflake",
+    ],
+    "expect_column_values_to_not_be_null-str quoted_upper_col": [
+        "databricks_sql",
+        "postgres",
+        "sqlite",
+    ],
+    'expect_column_values_to_not_be_null-str "quoted_upper_col"': [
+        "postgres",
+        "snowflake",
+    ],
+    "expect_column_values_to_not_be_null-str QUOTED_UPPER_COL": [
+        "postgres",
+    ],
+    'expect_column_values_to_not_be_null-str "QUOTED_UPPER_COL"': [
+        "postgres",
+        "snowflake",
+    ],
+}
+
+# expect failures for these column names
+# NOTE: the expectation must fail without a raised_exception
+EXPECTED_FAILURE: Final[dict[ColNameParamId, list[DatabaseType]]] = {
+    # TODO: add these for sqlite and databricks
+    # DDL: unquoted_lower_col ----------------------------------
+    'str "unquoted_lower_col"': ["snowflake"],
+    'str "UNQUOTED_LOWER_COL"': ["postgres"],
+    # DDL: UNQUOTED_UPPER_COL ----------------------------------
+    'str "unquoted_upper_col"': ["snowflake"],
+    'str "UNQUOTED_UPPER_COL"': ["postgres"],
+    # DDL: "quoted_lower_col"-----------------------------------
+    "str quoted_lower_col": ["postgres", "snowflake"],
+    "str QUOTED_LOWER_COL": ["postgres", "snowflake"],
+    'str "QUOTED_LOWER_COL"': ["postgres", "snowflake"],
+    # DDL: "QUOTED_UPPER_COL" ----------------------------------
+    "str quoted_upper_col": ["postgres"],
+    'str "quoted_upper_col"': ["postgres", "snowflake"],
+    "str QUOTED_UPPER_COL": ["postgres"],
```

Comment:
Could someone check my math on these expected failures?

Reply:
`str quoted_lower_col` should not fail for `postgres`
`str QUOTED_LOWER_COL` should not fail for `postgres`

Reply:
https://github.com/great-expectations/great_expectations/pull/8757#discussion_r1341393595